src/Hl7.Fhir.Base/Rest/BaseFhirClient.cs

```diff
@@ -632,7 +632,7 @@ public virtual void Delete(string resourceType, SearchParams condition)
         public Task<TResource?> PatchAsync<TResource>(SearchParams condition, Parameters patchParameters, CancellationToken? ct = null) where TResource : Resource
         {
             var tx = new TransactionBuilder(Endpoint);
-            var resourceType = Inspector.GetFhirTypeNameForType(typeof(TResource));
+            var resourceType = Inspector.GetFhirTypeNameForType(typeof(TResource)) ?? throw new ArgumentException($"{typeof(TResource).Name} is not recognized as a valid FHIR resource.");
```

Comment:
use `typeNameOrDie()`

Reply:
Ah ok. I was not aware of this. It was a proposal of Visual Studio to do this, and I agreed...